### ISSUE TITLE
Userstyles should be injected after body

### ISSUE THREAD

**Issue by mxmou** [ScratchAddons/ScratchAddons#701]
**Is your feature request related to a problem? Please describe.**

Styles for the footer on 2.0-styled pages are duplicated - they appear both in [`main.css`](https://cdn.scratch.mit.edu/scratchr2/static/css/main.css) and in the footer element itself. This means addons can't override them without using more specific selectors or `!important`.

**Describe the solution you'd like**

Userstyles should be injected when the `<body>` element already exists.

**Describe alternatives you've considered**

Userstyles could use more specific selectors or `!important`.

**Comment by WorldLanguages** [comment]
This conflicts with #328. Having to wait until the body has loaded to inject the styles would likely make the flickering return (as always, it depends a lot on the circumstances, of course). Maybe we could have a "load userstyle on full load" option? But it would mess with the ordering. What do you suggest?

**Comment by mxmou** [comment]
@WorldLanguages Could userstyles be moved after `<body>` when it loads?

**Comment by WorldLanguages** [comment]
@mxmou Yeah, that could be done. I'll experiment later. In the meantime, I suggest you add some `!important`s, and make a comment mentioning this issue.

**Comment by mxmou** [comment]
There is now a lot of CSS that uses `!important` to override CSS in the `<body>` and it isn't a big problem. Is this still necessary?

**Comment by WorldLanguages** [comment]
You're right, I think it's fine as long as we play close attention to usage of `!important`. Do you have any unwritten rules on what situations it should and shouldn't be used? Would be cool to have a page in the docs about userstyle conventions.

**Comment by mxmou** [comment]
> You're right, I think it's fine as long as we play close attention to usage of `!important`. Do you have any unwritten rules on what situations it should and shouldn't be used? Would be cool to have a page in the docs about userstyle conventions.

The only "rule" is that if it's possible to change the order of the styles or addons, that's better than using `!important`.

**Comment by WorldLanguages** [comment]
> Do you have any unwritten rules on what situations it should and shouldn't be used? Would be cool to have a page in the docs about userstyle conventions.

See https://github.com/ScratchAddons/website-v2/pull/311

**Comment by mxmou** [comment]
Closing because very few addons are affected by this, and those that are can simply use specificity or `!important` to override the styles.
